idaes/surrogate/tests/test_alamopy.py

```diff
@@ -0,0 +1,350 @@
+        "ZLABELS z1\n"
+        "XMIN 0 0\n"
+        "XMAX 5 10\n"
+        "NDATA 4\n"
+        "NVALDATA 0\n\n"
+        "linfcns 1\n"
+        "constant 1\n"
+        "maxtime 1000.0\n"
+        "numlimitbasis 1\n\n"
+        "TRACE 1\n"
+        "TRACEFNAME foo.bar\n"
+        "BEGIN_DATA\n"
+        "1 5 10\n"
+        "2 6 20\n"
+        "3 7 30\n"
+        "4 8 40\n"
+        "END_DATA\n")
+
+
+@pytest.mark.unit
+def test_writer_vdata(alm_obj):
+    alm_obj._vdata_in = np.array([[2.5], [6.5]])
+    alm_obj._vdata_out = np.array([25], ndmin=2)
+
+    stream = io.StringIO()
+    alm_obj.write_alm_to_stream(stream=stream)
+
+    assert stream.getvalue() == (
+        "# IDAES Alamopy input file\n"
+        "NINPUTS 2\n"
+        "NOUTPUTS 1\n"
+        "XLABELS x1 x2\n"
+        "ZLABELS z1\n"
+        "XMIN 0 0\n"
+        "XMAX 5 10\n"
+        "NDATA 4\n"
+        "NVALDATA 1\n\n"
+        "linfcns 1\n"
+        "constant 1\n"
+        "maxtime 1000.0\n"
+        "numlimitbasis 1\n\n"
+        "TRACE 1\n\n"
+        "BEGIN_DATA\n"
+        "1 5 10\n"
+        "2 6 20\n"
+        "3 7 30\n"
+        "4 8 40\n"
+        "END_DATA\n"
+        "\nBEGIN_VALDATA\n"
+        "2.5 6.5 25\n"
+        "END_VALDATA\n")
+
+
+@pytest.mark.unit
+def test_writer_full_config(alm_obj):
+    alm_obj.config.xfactor = [1.2, 1.6]
+    alm_obj.config.xscaling = True
+    alm_obj.config.scalez = True
+
+    alm_obj.config.monomialpower = [1, 2]
+    alm_obj.config.multi2power = [1, 2, 3]
+    alm_obj.config.multi3power = [1, 2, 3, 4]
+    alm_obj.config.ratiopower = [1, 2, 3, 4, 5]
+
+    alm_obj.config.constant = True
+    alm_obj.config.linfcns = False
+    alm_obj.config.expfcns = True
+    alm_obj.config.logfcns = False
+    alm_obj.config.sinfcns = True
+    alm_obj.config.cosfcns = False
+    alm_obj.config.grbfcns = True
+    alm_obj.config.rbfparam = 7
+
+    alm_obj.config.modeler = Modelers.AICc
+    alm_obj.config.builder = True
+    alm_obj.config.backstepper = False
+    alm_obj.config.convpen = 42
+    alm_obj.config.screener = Screener.SIS
+    alm_obj.config.ncvf = 4
+    alm_obj.config.sismult = 5
+
+    alm_obj.config.maxiter = -50
+    alm_obj.config.maxtime = 500
+    alm_obj.config.datalimitterms = True
+    alm_obj.config.maxterms = [-1, 2]
+    alm_obj.config.minterms = [2, -1]
+    alm_obj.config.numlimitbasis = False
+    alm_obj.config.exclude = [1, 0]
+    alm_obj.config.ignore = 1
+    alm_obj.config.xisint = [1, 0]
+    alm_obj.config.zisint = 0
+
+    alm_obj.config.tolrelmetric = 8.1
+    alm_obj.config.tolabsmetric = 9.2
+    alm_obj.config.tolmeanerror = 10.3
+    alm_obj.config.tolsse = 11.4
+
+    alm_obj.config.mipoptca = 12.5
+    alm_obj.config.mipoptcr = 13.6
+    alm_obj.config.linearerror = False
+    alm_obj.config.GAMS = "foo"
+    alm_obj.config.GAMSSOLVER = "bar"
+    alm_obj.config.solvemip = False
+
+    stream = io.StringIO()
+    alm_obj.write_alm_to_stream(stream=stream)
+
+    assert stream.getvalue() == (
```

Comment:
This test should be not sensitive to things that Alamo also doesn't care about, like the order of keywords in the file. Suggest doing a comparison between `set(stream.getvalue().split("\n")` and a set constant with the strings ` {""# IDAES Alamopy input file", "NINPUTS 2", "NOUTPUTS 1", ...}`. You may want to do a partial match that just the strings in the set constant are in the Alamopy result, so later additions don't break the test.

Reply:
I am on the fence with that one - whilst ALAMO doesn't care about the order (with a couple of minor exceptions), the order *is* indicative of how the writer is working, I.e. if the order changes then something changed in the writer that maybe we should know about.